src/DateUtils.ts

```diff
@@ -18,90 +18,66 @@ limitations under the License.
 
 import { Optional } from "matrix-events-sdk";
 
-import { _t } from "./languageHandler";
+import { _t, getUserLanguage } from "./languageHandler";
 
-function getDaysArray(): string[] {
-    return [_t("Sun"), _t("Mon"), _t("Tue"), _t("Wed"), _t("Thu"), _t("Fri"), _t("Sat")];
-}
-
-function getMonthsArray(): string[] {
-    return [
-        _t("Jan"),
-        _t("Feb"),
-        _t("Mar"),
-        _t("Apr"),
-        _t("May"),
-        _t("Jun"),
-        _t("Jul"),
-        _t("Aug"),
-        _t("Sep"),
-        _t("Oct"),
-        _t("Nov"),
-        _t("Dec"),
-    ];
-}
+const MILLIS_IN_DAY = 86400000;
 
-function pad(n: number): string {
-    return (n < 10 ? "0" : "") + n;
+export function getDaysArray(weekday: Intl.DateTimeFormatOptions["weekday"] = "short"): string[] {
+    const now = new Date();
+    const { format } = new Intl.DateTimeFormat(getUserLanguage(), { weekday });
+    return [...Array(7).keys()].map((day) => format(now.getTime() - (now.getDay() - day) * MILLIS_IN_DAY));
 }
 
-function twelveHourTime(date: Date, showSeconds = false): string {
-    let hours = date.getHours() % 12;
-    const minutes = pad(date.getMinutes());
-    const ampm = date.getHours() >= 12 ? _t("PM") : _t("AM");
-    hours = hours ? hours : 12; // convert 0 -> 12
-    if (showSeconds) {
-        const seconds = pad(date.getSeconds());
-        return `${hours}:${minutes}:${seconds}${ampm}`;
-    }
-    return `${hours}:${minutes}${ampm}`;
+function getMonthsArray(month: Intl.DateTimeFormatOptions["month"] = "short"): string[] {
+    const { format } = new Intl.DateTimeFormat(getUserLanguage(), { month });
+    return [...Array(12).keys()].map((m) => format(new Date(Date.UTC(2021, m))));
 }
 
-export function formatDate(date: Date, showTwelveHour = false): string {
+export function formatDate(date: Date, showTwelveHour = false, locale?: string): string {
+    const _locale = locale ?? getUserLanguage();
     const now = new Date();
-    const days = getDaysArray();
-    const months = getMonthsArray();
     if (date.toDateString() === now.toDateString()) {
-        return formatTime(date, showTwelveHour);
+        return formatTime(date, showTwelveHour, _locale);
     } else if (now.getTime() - date.getTime() < 6 * 24 * 60 * 60 * 1000) {
```

Comment:
I assume you decoded this as part of this work. It would have been nice of you not to make me do it again.

```suggestion
    } else if (now.getTime() - date.getTime() < 6 * 24 * 60 * 60 * 1000) {
        // Time is less than 6 days into the future. Use a short time XXX wtf how does this actually differ from the other format
```